src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -233,6 +236,26 @@ EmberAfStatus OnOffServer::setOnOffValue(chip::EndpointId endpoint, chip::Comman
                 status = ModeSelect::Attributes::CurrentMode::Set(endpoint, onMode.Value());
             }
         }
+#endif
+#ifdef EMBER_AF_PLUGIN_MODE_BASE
+        // If OnMode is not a null value, then change the current mode to it.
+        for (const auto & element : ModeBase::Instance::ModeBaseAliasesInstanceMap)
+        {
+            auto modeBaseAlias = element.second;
+            if (modeBaseAlias->GetEndpointId() == endpoint)
+            {
+                if (modeBaseAlias->HasFeature(ModeBase::Feature::kOnOff))
+                {
+                    ModeBase::Instance * modeBaseInstance               = element.second;
+                    ModeBase::Attributes::OnMode::TypeInfo::Type onMode = modeBaseInstance->GetOnMode();
+                    if (!onMode.IsNull())
+                    {
+                        ChipLogProgress(Zcl, "Changing Current Mode to %x", onMode.Value());
+                        modeBaseInstance->UpdateCurrentMode(onMode.Value());
+                    }
+                }
+            }
+        }
```

Comment:
This code is identical to the other block above. The only local value used are `endpoint` and `onMode.value()`. Suggest factoring this common behavior in a top-level anonymous namespace method.

Reply:
Resolved [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/7312989283d52cf4c9b1f1655b970da77d58d7bb).